### PR TITLE
Adopt Jackson 2.15

### LINKED ISSUE #5283: don't uptake jackson 2.15.0+ without taking into account its new StreamReadConstraints
Jackson 2.15 (v2.15.0-rc1 is out) has a default set of limits that it applies to inputs. Input Files that breach those limits will cause parse exceptions.

jersey-media-json-jackson will need to consider if those limits are ok or if you need to provide a way to configure the Jackson StreamReadConstraints.

* https://javadoc.io/static/com.fasterxml.jackson.core/jackson-core/2.15.0-rc1/com/fasterxml/jackson/core/StreamReadConstraints.html

### PR DESCRIPTION
**Pull request by jansupol**

### CODE REVIEW

**Review by senivam — APPROVED** [review]

**Review by senivam — APPROVED** [review]

**Review by jbescos — APPROVED** [review]

File: docs/src/main/docbook/appendix-properties.xml [hunk 1]
```diff
@@ -1159,6 +1159,118 @@
...
+                        <entry>&jersey.message.MessageProperties.JSON_MAX_STRING_LENGTH;</entry>
+                        <entry><literal>jersey.config.json.string.length</literal></entry>
+                        <entry>
+                            <para>
+                                The default value is not set and the JSON provider default maximum value is used.
+                            </para>
+                            <para>
+                                If supported by Jackson provider, the default value can differ for each Jackson version. For instance,
+                                Jackson 14 does not support this setting and the default value is <literal>Integer#MAX_VALUE</literal>,
+                                Jackson 15.0 has the default value 5_000_000, Jackson 15.2 has the default value 20_000_000.
```
Comment (pjfanning) [inline docs/src/main/docbook/appendix-properties.xml:1233]: The Jackson versions are 2.14 and 2.15 - there is no 14 or 15. The 20m limit was brought in in Jackson 2.15.1.

See https://github.com/FasterXML/jackson-core/blob/2.16/release-notes/VERSION-2.x

File: docs/src/main/docbook/migration.xml [hunk 2]
```diff
@@ -28,6 +28,18 @@
          xml:id="migration">
     <title>Migration Guide</title>
 
+    <section xml:id="mig-2.40">
+        <title>Migrating from Jersey 2.40 to 2.41</title>
+        <section xml:id="mig-2.40-breaking-changes">
+            <title>Changes in Jackson</title>
+            <para>
+                Jersey 2.41 starts to support Jackson 15 which comes with default limitations for the length of parsed text,
```
Comment (pjfanning) [inline docs/src/main/docbook/migration.xml:36]: again, no Jackson v15 has ever been released

File: media/json-jackson/src/main/resources/org/glassfish/jersey/jackson/localization.properties [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+# This Source Code may also be made available under the following Secondary
+# Licenses when the conditions for such availability set forth in the
+# Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+# version 2 with the GNU Classpath Exception, which is available at
+# https://www.gnu.org/software/classpath/license.html.
+#
+# SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+#
+
+error.jackson.streamreadconstraints=Error setting StreamReadConstraints: {0}. Possibly not Jackson 15?
```
Comment (pjfanning) [inline media/json-jackson/src/main/resources/org/glassfish/jersey/jackson/localization.properties:17]: again, no Jackson v15 has ever been released

File: tests/integration/jackson-14/pom.xml [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        <artifactId>project</artifactId>
+        <groupId>org.glassfish.jersey.tests.integration</groupId>
+        <version>2.41-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>jackson-14</artifactId>
+    <name>jersey-compatibility-jackson14</name>
+    <description>
+        Controls the backward compatibility with Jackson 14 environment
```
Comment (pjfanning) [inline tests/integration/jackson-14/pom.xml:33]: again, no Jackson v14 has ever been released

File: tests/integration/pom.xml [hunk 5]
```diff
@@ -44,6 +44,7 @@
         <module>j-376</module>
         <module>j-441</module>
         <module>j-59</module>
+        <module>jackson-14</module>
```
Comment (pjfanning) [inline tests/integration/pom.xml:47]: again, no Jackson v14 has ever been released

File: docs/src/main/docbook/migration.xml (hunk 2 above)
Comment (jansupol) [reply docs/src/main/docbook/migration.xml:36]: Right, it is 2.15

**Pull request merged**